includes/compat/class-wc-stripe-subscriptions-helper.php

```diff
@@ -63,16 +85,66 @@ public static function get_some_detached_subscriptions() {
 						'customer_id'               => $subscription->get_meta( '_stripe_customer_id' ),
 						'change_payment_method_url' => $subscription->get_change_payment_method_url(),
 					];
-					if ( count( $detached_subscriptions ) >= 5 ) {
-						break;
-					}
 				}
 			}
 		}
 
 		// Cache the result for a day.
-		set_transient( self::DETACHED_SUBSCRIPTIONS_TRANSIENT_KEY, $detached_subscriptions, DAY_IN_SECONDS );
+		WC_Stripe_Database_Cache::set( self::DETACHED_SUBSCRIPTIONS_CACHE_KEY, $detached_subscriptions, DAY_IN_SECONDS );
```

Comment:
As noted above, we should store the value of `$limit` alongside the cached data.
```suggestion
		WC_Stripe_Database_Cache::set( self::DETACHED_SUBSCRIPTIONS_CACHE_KEY, array( 'limit' => $limit, 'subscriptions' => $detached_subscriptions ), DAY_IN_SECONDS );
```